core/src/main/java/nu/yona/server/goals/service/GoalDto.java

```diff
@@ -45,7 +45,7 @@
 	protected GoalDto(UUID id, Optional<LocalDateTime> creationTime, Optional<LocalDateTime> endTime, UUID activityCategoryId,
 			boolean mandatory)
 	{
-		this.id = id;
+		this.id = (id == null) ? new UUID(0, 0) : id;
```

Comment:
Isn't it possible to make `id` an `Optional<UUID>`?

Reply:
I played with that. It gives a slightly more elegant solution, but cannot be done because ``Optional`` is not serializable. And ``GoalDto`` needs to be serializable, as it's cached in Hazelcast.

Reply:
OK, we could add a comment about that to explain the choice. But can't we make the public `getGoalId` an `Optional<UUID>` and keep the serialization details private?

Reply:
It would formally be right to make it optional when the zero UUID is set, but practically, it's a lot of clutter for no value. The scenario where we do not have a UUID is limited to the case where a new goal DTO is serialized to us. That scenario does not retrieve the ID from the DTO but generates a new one. All scenarios that take the ID from the DTO can rely on the availability of the ID.

Given that, I'd rather explain this story in a comment in ``GoalDto``.